lang/attribute/event/src/lib.rs

```diff
@@ -81,6 +81,35 @@ pub fn emit(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
     })
 }
 
+#[proc_macro]
+pub fn emit_cpi(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let tuple = parse_macro_input!(input as syn::ExprTuple);
+
+    let elems = tuple.elems;
+    if elems.len() != 2 {
+        panic!("Expected a tuple with exactly two elements.");
+    }
+
+    let first = &elems[0];
+    let second = &elems[1];
+
+    proc_macro::TokenStream::from(quote! {
+        let program_info: &anchor_lang::solana_program::account_info::AccountInfo = &#first;
+
+        let __disc = crate::event::EVENT_IX_TAG_LE;
+        let __inner_data: Vec<u8> = anchor_lang::Event::data(&#second);
+        let ix_data: Vec<u8> = __disc.into_iter().chain(__inner_data.into_iter()).collect();
+
+        let ix = anchor_lang::solana_program::instruction::Instruction::new_with_bytes(
+            program_info.key.clone(),
+            ix_data.as_ref(),
+            vec![anchor_lang::solana_program::instruction::AccountMeta::new_readonly(*program_info.key, false)],
+        );
+        anchor_lang::solana_program::program::invoke(&ix, &[program_info.clone()])
```

Comment:
No need to clone here

```suggestion
        anchor_lang::solana_program::program::invoke(&ix, &[program_info])
```

Reply:
Fixed